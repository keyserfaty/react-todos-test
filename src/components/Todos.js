import React from 'react'
import TodosInputContainer from '../containers/TodosInputContainer'

const Todos = ({ todos, onAddTodoClick }) => (
  <div id="todos-container">
    <TodosInputContainer onAddTodoClick={onAddTodoClick} />
    <div id="todos-list">
      <ul>
        { todos.map((todo, i) => (
          <li key={i}>{todo}</li>
        )) }
      </ul>
    </div>
  </div>
);

export default Todos
