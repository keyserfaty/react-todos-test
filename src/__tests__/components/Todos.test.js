import React from 'react'
import renderer from 'react-test-renderer'

import Todos from '../../components/Todos'

it('should render a todos component', () => {
  const todos = [
    'todo 1',
    'todo 2'
  ];

  const tree = renderer.create(
    <Todos todos={todos} />
  ).toJSON();

  expect(tree).toMatchSnapshot()
});
