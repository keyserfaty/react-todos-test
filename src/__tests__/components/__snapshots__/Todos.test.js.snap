exports[`test should render a todos component 1`] = `
<div
  id="todos-container">
  <div
    id="todos-input">
    <input
      onChange={[Function]}
      type="text"
      value="" />
    <button
      onClick={[Function]}>
      Add todo!
    </button>
  </div>
  <div
    id="todos-list">
    <ul>
      <li>
        todo 1
      </li>
      <li>
        todo 2
      </li>
    </ul>
  </div>
</div>
`;
